README.md

```diff
@@ -186,6 +124,69 @@ print("Updated params:", m.get_params())
+generality, a JAX program is bounded only by what can run in the host Python
+interpreter. What we are looking for is a simpler definition that allows a
+useful set of standalone programs to be constructed and that meshes well with
+the programming model employed by typical JAX developers.
+
+The components that are the most interesting towards this end are:
+
+* `jax.jit` functions: stateless functions mapping inputs to outputs, natively
+  represented as JAXPR IR and universally convertible to MHLO IR.
+* Multi-dimensional arrays very similar to numpy arrays, with extensions to
+  enable tracing and distribution.
+* Trees of arrays and primitives (a structured generalization of nested dicts,
+  lists and tuples).
+* "AbstractValues" mirroring all concrete data types, facilitating symbolic
+  tracing and descriptions of program elements.
+
+There are of course many more details than this, but these are the components
+we will assemble.
+
+## What is an IREE program?
+
+An IREE program is:
+
+* Defined by an MLIR module containing:
+  * A name, which can be used for cross-referencing between modules. All modules
+    loaded into a single runtime instance must have a unique name.
+  * Zero or more globals. Globals can contain any scalar or multi-dimensional
+    array of any supported type (IREE has an open type system and supports
+    more than this but these types are relevant to JAX). Globals can be:
+      * Loaded, yielding their current value
+      * Stored, updating the value
+      * Mutable or immutable
+      * Initialized, either via a constant or via special initializer functions
+        that can do whatever they want at module load time
+  * One or more public functions, each of which accepts and returns scalars and
+    buffers representationing multi-dimensional arrays. IREE functions can
+    do a lot of things, but relevant to JAX, they can include:
+      * A graph of MHLO operations (the native IR of JAX jitted functions)
+      * Operations and types from
+        [IREE's input dialect](https://github.com/google/iree/blob/main/llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/Input/InputOps.td)
+      * Operations from low level MLIR dialects (for programs coming from a high
+        level IR like MHLO, these are typically not used, but they provide
+        the generality needed for more advanced cases such as implementing
+        full language compilers):
+        * MLIR [arith](https://mlir.llvm.org/docs/Dialects/ArithmeticOps/) dialect
+        * MLIR [linalg](https://mlir.llvm.org/docs/Dialects/Linalg/) dialect
+        * MLIR [math](https://mlir.llvm.org/docs/Dialects/MathOps/) dialect
+        * MLIR [std](https://mlir.llvm.org/docs/Dialects/Standard/) dialect (being phased out)
+        * MLIR [tensor](https://mlir.llvm.org/docs/Dialects/TensorOps/) dialect
+* Compiled by the `ireec` compiler, targeting a number or architectures and
```

Comment:
Is the final artifact generated by the ireec compiler the "IREE VM Program"?

Reply:
That is the final MLIR form (i.e. a module in the `VM` dialect). It is then translated to a flatbuffer (vmfb), a C program, or (soon) directly to LLVM IR.